From d8438e6070d7cf1c6fdeb04f887964a8b1f75669 Mon Sep 17 00:00:00 2001
From: samaga Prasanna Krishna <samaga.krishna@intel.com>
Date: Thu, 28 Mar 2013 16:42:10 +0530
Subject: [PORT FROM MAIN] [CTP-OFFLOAD] avoid getRenderPositon to HAL

BZ: 93519

Avoid calling getRenderPosition call to HAL when the audio
playback has completed. AT_EOS flag is set in onStreamDone
function. Using this flag to avoid calling the render postion
in HAL.

This is a work around fix for this issue to avoid the crash.

Change-Id: I25a1b5ff206aa64435c9c35949e64f67f5ae7789
Orig-Change-Id: I4c6ed1bc99d58e124029b1b53f48a2d4e864a946
Signed-off-by: samaga Prasanna Krishna <samaga.krishna@intel.com>
Reviewed-on: http://android.intel.com:8080/98897
Reviewed-by: R Padmanaban, Jayashree <jayashree.r.padmanaban@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: Gupta, ArvindX K <arvindx.k.gupta@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index e216545..fbe6656 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1320,7 +1320,7 @@ status_t AwesomePlayer::getPosition(int64_t *positionUs) {
             && (mAudioPlayer == NULL || !(mFlags & VIDEO_AT_EOS))) {
         Mutex::Autolock autoLock(mMiscStateLock);
         *positionUs = mVideoTimeUs;
-    } else if (mAudioPlayer != NULL) {
+    } else if (mAudioPlayer != NULL && !(mFlags & AT_EOS)) {
         *positionUs = mAudioPlayer->getMediaTimeUs();
     } else {
         *positionUs = 0;
-- 
1.7.4.1

